src/main/kotlin/fi/metatavu/metaform/server/rest/UsersApi.kt

```diff
@@ -0,0 +1,146 @@
+            val upnNumber = user.displayName!!.split(" ")
+
+            if (upnNumber.size == 3) {
+                if (existingUsers.any { it.username!!.contains(upnNumber[2]) }) {
+                    return createConflict(createDuplicatedMessage(UPN_NUMBER))
+                }
+            }
+
+            val createdUser = usersController.createUser(user)
+
+            return createOk(userTranslator.translate(createdUser))
+        } catch (e: Exception) {
+            return createInternalServerError(e.localizedMessage)
+        }
+    }
+
+    override fun deleteUser(userId: UUID): Response {
+        loggedUserId ?: return createForbidden(UNAUTHORIZED)
+
+        if (!isRealmSystemAdmin) {
+            return createForbidden(UNAUTHORIZED)
+        }
+
+        metaformKeycloakController.findUserById(userId)
+            ?: return createNotFound(createNotFoundMessage(USER, userId))
+
+        metaformKeycloakController.deleteUser(userId)
+
+        return createNoContent()
+    }
+
+    override fun findUser(userId: UUID): Response {
+        loggedUserId ?: return createForbidden(UNAUTHORIZED)
+
+        if (!isMetaformAdminAny) {
+            return createForbidden(UNAUTHORIZED)
+        }
+
+        val foundUser = metaformKeycloakController.findUserById(userId)
+            ?: return createNotFound(createNotFoundMessage(USER, userId))
+
+        return createOk(userTranslator.translate(foundUser))
+    }
+
+    override fun listUsers(search: String?, firstResult: Int?, maxResults: Int?): Response {
+        loggedUserId ?: return createForbidden(UNAUTHORIZED)
+
+        if (!isMetaformAdminAny) {
+            return createForbidden(UNAUTHORIZED)
+        }
+
+        val foundUsers = metaformKeycloakController.findUsersBySearchParam(
+            search = search,
+            firstResult = firstResult,
+            maxResults = maxResults
+        )
+            .map { metaformKeycloakController.findUserById(UUID.fromString(it.id!!)) }
+            .map { userTranslator.translate(it!!) }
+            .toMutableList()
+
+        val cardAuthKeycloakUsers = cardAuthKeycloakController.findUsersBySearchParam(
+            search = search,
+            firstResult = firstResult,
+            maxResults = maxResults
+        )
+            .toMutableList()
+            .filter { cardAuthKeycloakUser ->
+                foundUsers.find { it.federatedIdentities?.get(0)?.userId == cardAuthKeycloakUser.id } == null
+            }
+            .map { userTranslator.translateCardAuthUserRepresentation(it) }
+
+        foundUsers.addAll(cardAuthKeycloakUsers)
+
+        var totalUsers = metaformKeycloakController.findUsersBySearchParam(
+            search = search,
+            maxResults = 1000
+        ).size
+
+        totalUsers += cardAuthKeycloakController.findUsersBySearchParam(
+            search,
+            maxResults = 1000
+        ).size
+
+        return createOk(
+            entity = foundUsers,
+            count = totalUsers
```

Comment:
Count in this case can be quite misleading, so I would remove it

Reply:
Removed